test/with_dummyserver/test_socketlevel.py

```diff
@@ -1133,11 +1145,12 @@ def request():
             finally:
                 pool.close()
 
-        with self.assertRaises(MaxRetryError) as cm:
+        with pytest.raises(MaxRetryError) as cm:
             request()
-        self.assertIsInstance(cm.exception.reason, SSLError)
+            assert isinstance(cm.exception.reason, SSLError)
```

Comment:
You also have two extra spaces here

Reply:
It is giving error if it is not in the block of with.

Reply:
Ah! Nice catch. That will teach me to read the documentation I link to. :) The syntax with pytest has changed. You need to use this form:

```python3
with pytest.raises(MaxRetryError) as excinfo:
    pool.request('GET', '/', retries=0)
assert isinstance(excinfo.value.reason, SSLError)
```

(excinfo.value instead of cm.exception)

Sorry about that!